stub-idp/src/main/java/uk/gov/ida/stub/idp/resources/idp/LoginPageResource.java

```diff
@@ -228,6 +249,65 @@ public Response postAuthnPending(
         return samlResponseRedirectViewFactory.sendSamlMessage(pendingResponse);
     }
 
+
+    private boolean sessionContainsUser(Optional<IdpSession> session) {
+        return session.isPresent() && session.get().getIdpUser().isPresent();
+    }
+
+    private boolean sessionHasIdaAuthnRequestFromHub(Optional<IdpSession> session) {
+        return session.isPresent() && session.get().getIdaAuthnRequestFromHub() != null;
+    }
+
+    private Response createSessionAndShowLoginForm(Idp idp, Optional<ErrorMessageType> errorMessage){
+
+        final IdpSession idpSession = new IdpSession(
+                new SessionId(UUID.randomUUID().toString()),
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                Optional.of(UUID.randomUUID()));
```

Comment:
Don't think we need the `singleIdpJourneyId` yet ... maybe this constructor only needs one parameter.